backend/api/Services/AreaService.cs

```diff
@@ -0,0 +1,247 @@
+﻿using Api.Controllers.Models;
+using Api.Database.Context;
+using Api.Database.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace Api.Services
+{
+    public interface IAreaService
+    {
+        public abstract Task<IEnumerable<Area>> ReadAll();
+
+        public abstract Task<Area?> ReadById(string id);
+
+        public abstract Task<IEnumerable<Area>> ReadByAsset(string asset);
+
+        public abstract Task<Area?> ReadByAssetAndName(string assetName, string name);
+
+        public abstract Task<Area> Create(CreateAreaQuery newArea);
+
+        public abstract Task<Area> Create(CreateAreaQuery newArea, List<Pose> safePositions);
+
+        public abstract Task<Area> Update(Area area);
+
+        public abstract Task<Area?> AddSafePosition(string asset, string name, SafePosition safePosition);
+
+        public abstract Task<Area?> Delete(string id);
+
+    }
+
+    [System.Diagnostics.CodeAnalysis.SuppressMessage(
+        "Globalization",
+        "CA1309:Use ordinal StringComparison",
+        Justification = "EF Core refrains from translating string comparison overloads to SQL"
+    )]
+    [System.Diagnostics.CodeAnalysis.SuppressMessage(
+        "Globalization",
+        "CA1304:Specify CultureInfo",
+        Justification = "Entity framework does not support translating culture info to SQL calls"
+    )]
+    public class AreaService : IAreaService
+    {
+        private readonly FlotillaDbContext _context;
+
+        public AreaService(FlotillaDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<IEnumerable<Area>> ReadAll()
+        {
+            return await GetAreas().ToListAsync();
+        }
+
+        private IQueryable<Area> GetAreas()
+        {
+            return _context.Areas.Include(a => a.SafePositions)
+                .Include(a => a.Deck).ThenInclude(d => d.Installation).ThenInclude(i => i.Asset);
```

Comment:
```suggestion
            return _context.Areas.Include(a => a.SafePositions)
                .Include(a => a.Deck).Include(d => d.Installation).Include(i => i.Asset);
```
Since the `Area` model now has all these included in a flat structure it should not need to request in a nested manner through `Deck`.